src/main/java/com/hierynomus/smbj/share/DiskShare.java

```diff
@@ -466,6 +465,152 @@ private boolean exists(String path, EnumSet<SMB2CreateOptions> createOptions) th
         }
     }
 
+    /***
+     * Rename(Move) Directory. Default will using the false for replaceIfExist option
+     * @param path the path to the original directory
+     * @param newFileName the new directory name/path after rename
+     * @throws TransportException
+     * @throws SMBApiException
+     */
+    public void renameDirectory(String path, String newFileName) throws TransportException, SMBApiException {
```

Comment:
Here I'm doubting whether we should implement `rename` operations on the `File` and `Directory` objects instead of the `rename(Directory|File)` methods on the DiskShare.

What are your thoughts on this?